arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -869,17 +867,67 @@ bool RocksDBMetaCollection::haveBufferedOperations() const {
 }
 
 std::size_t RocksDBMetaCollection::revisionTreeDepth() const {
-  return 6;  // TODO make variable
+  std::size_t const treeCount = _revisionTree ? _revisionTree->count() : 0;
+  std::size_t bufferInserts = 0;
+  std::size_t bufferRemovals = 0;
+  {
+    std::unique_lock<std::mutex> guard(_revisionBufferLock);
+
+    rocksdb::SequenceNumber ignoreSeq = 0;
+    if (!_revisionTruncateBuffer.empty()) {
+      auto it = _revisionTruncateBuffer.rbegin();
+      if (it != _revisionTruncateBuffer.rend()) {
+        ignoreSeq = *it;
+      }
+    }
+
+    for (auto& pair : _revisionInsertBuffers) {
+      if (pair.first <= ignoreSeq) {
+        continue;
+      }
+
+      bufferInserts += pair.second.size();
+    }
+
+    for (auto& pair : _revisionRemovalBuffers) {
+      if (pair.first <= ignoreSeq) {
+        continue;
+      }
+
+      bufferRemovals += pair.second.size();
+    }
+  }
+
+  std::size_t const treeDepth = _revisionTree ? _revisionTree->maxDepth() : 3;
+
+  if (bufferInserts > bufferRemovals) {
+    std::size_t const count = treeCount + bufferInserts - bufferRemovals;
+    std::size_t targetDepth = 3;
```

Comment:
Should `targetDepth` start with the value of `treeDepth`?
I guess otherwise it would be possible to make this function return a smaller tree depth that the tree's current size.
```suggestion
    std::size_t targetDepth = 3;
```